Makefile.uk

```diff
@@ -0,0 +1,233 @@
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+################################################################################
+# Library registration
+################################################################################
+$(eval $(call addlib_s,libsodium,$(CONFIG_LIBSODIUM)))
+
+################################################################################
+# Original sources
+################################################################################
+LIBSODIUM_VERSION = 1.0.18
+LIBSODIUM_LIBRARY_VERSION_MAJOR = 10
+LIBSODIUM_LIBRARY_VERSION_MINOR = 3
+LIBSODIUM_URL = https://github.com/jedisct1/libsodium/archive/refs/tags/$(LIBSODIUM_VERSION).zip
+LIBSODIUM_SUBDIR = libsodium-$(LIBSODIUM-VERSION)
+LIBSODIUM_PATCHDIR = $(LIBSODIUM_BASE)/patches
+$(eval $(call fetch,libsodium,$(LIBSODIUM_URL)))
+$(eval $(call patch,libsodium,$(LIBSODIUM_PATCHDIR),$(LIBSODIUM_SUBDIR)))
+
+################################################################################
+# Helpers
+################################################################################
+LIBSODIUM_EXTRACTED = $(LIBSODIUM_ORIGIN)/libsodium-$(LIBSODIUM_VERSION)
+
+################################################################################
+# Library includes
+################################################################################
+LIBSODIUM_COMMON_INCLUDES-y     += -I$(LIBSODIUM_EXTRACTED)/src/libsodium/include/sodium
+LIBSODIUM_CINCLUDES             += $(LIBSODIUM_COMMON_INCLUDES-y)
+LIBSODIUM_CXXINCLUDES           += $(LIBSODIUM_COMMON_INCLUDES-y)
+
+CINCLUDES-$(CONFIG_LIBSODIUM)   += -I$(LIBSODIUM_EXTRACTED)/src/libsodium/include
+CXXINCLUDES-$(CONFIG_LIBSODIUM) += -I$(LIBSODIUM_EXTRACTED)/src/libsodium/include
+
+################################################################################
+# Global flags
+################################################################################
+LIBSODIUM_SUPPRESS_FLAGS += -fno-strict-aliasing -fno-strict-overflow
+
+LIBSODIUM_CFLAGS-y   += $(LIBSODIUM_SUPPRESS_FLAGS)
+LIBSODIUM_CXXFLAGS-y += $(LIBSODIUM_SUPPRESS_FLAGS)
+
+LIBCRYPTO_CFLAGS-y   += $(LIBSODIUM_SUPPRESS_FLAGS)
+LIBCRYPTO_CXXFLAGS-y += $(LIBSODIUM_SUPPRESS_FLAGS)
+
+# Extracted from original build by running configure --minimal and removing the following:
+#
+# -DHAVE_SYS_RANDOM_H=1 -DHAVE_ALLOCA_H=1 -DHAVE_ALLOCA=1 -DHAVE_MMAP=1 -DHAVE_MLOCK=1 -DHAVE_MADVISE=1 -DHAVE_MPROTECT=1
+# -DHAVE_GETRANDOM=1 -DHAVE_GETENTROPY=1 -DHAVE_POSIX_MEMALIGN=1 -DHAVE_GETPID=1 -DHAVE_NANOSLEEP=1 -DHAVE_EXPLICIT_BZERO=1 
+#
+LIBSODIUM_DEFINES += -DPACKAGE_NAME=\"libsodium\" \
+                     -DPACKAGE_TARNAME=\"libsodium\" \
+                     -DPACKAGE_VERSION=\"$(LIBSODIUM_VERSION)\" \
+                     -DPACKAGE_STRING=\"libsodium\ 1.0.18\" \
```

Comment:
Can you please add the version number variable here too? :)

Reply:
Ooops, fixed now